source/extensions/filters/http/health_check/health_check.h

```diff
@@ -27,9 +27,12 @@ class HealthCheckCacheManager {
 public:
   HealthCheckCacheManager(Event::Dispatcher& dispatcher, std::chrono::milliseconds timeout);
 
-  Http::Code getCachedResponseCode() { return last_response_code_; }
-  void setCachedResponseCode(Http::Code code) {
+  std::pair<Http::Code, bool> getCachedResponseCode() {
```

Comment:
nit: Can you rename `getCachedResponseCode()` and `setCachedResponseCode()` to something more descriptive now that we are doing more stuff? s/Code/Data or something like that?